> I have become aware that this is a GCC extension, which I was not aware of previously. However, I'm not seeing what this has to do with C++20/GCC 11 support as the extension seems to have existed for quite some time so it should compile fine on GCC 11? What error are you getting?

The error message:
```
/app/x86_64-unknown-linux-gnu/bin/x86_64-unknown-linux-gnu-g++ -DVAPOR_SYSTEM_DATA_PATH=\"/app/share/VapoR\" -Dvrclient_EXPORTS -I/run/build/vapor/src/. -O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=2 -idirafter/usr/include -std=gnu++20 -fPIC -MD -MT src/CMakeFiles/vrclient.dir/backend/openxr.cpp.o -MF src/CMakeFiles/vrclient.dir/backend/openxr.cpp.o.d -o src/CMakeFiles/vrclient.dir/backend/openxr.cpp.o -c /run/build/vapor/src/backend/openxr.cpp
/run/build/vapor/src/backend/openxr.cpp: In constructor 'OpenXR::Session::Session(const OpenXR::Instance&, XrSystemId)':
/run/build/vapor/src/backend/openxr.cpp:100:103: error: taking address of temporary array
  100 |     GLXFBConfig* fbConfigs = glXChooseFBConfig(this->xDisplay, DefaultScreen(this->xDisplay), (int[]) {
      |                                                                                                       ^
  101 |         GLX_X_RENDERABLE, True,
      |         ~~~~~~~~~~~~~~~~~~~~~~~                                                                        
  102 |         GLX_DRAWABLE_TYPE, GLX_PBUFFER_BIT,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                            
  103 |         GLX_RENDER_TYPE, GLX_RGBA_BIT,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                 
  104 |         GLX_RED_SIZE, 8,
      |         ~~~~~~~~~~~~~~~~                                                                               
  105 |         GLX_GREEN_SIZE, 8,
      |         ~~~~~~~~~~~~~~~~~~                                                                             
  106 |         GLX_BLUE_SIZE, 8,
      |         ~~~~~~~~~~~~~~~~~                                                                              
  107 |         GLX_ALPHA_SIZE, 8,
      |         ~~~~~~~~~~~~~~~~~~                                                                             
  108 |         GLX_DEPTH_SIZE, 24,
      |         ~~~~~~~~~~~~~~~~~~~                                                                            
  109 |         None
      |         ~~~~                                                                                           
  110 |     }, &fbConfigsCounts);
      |     ~                                                                                                  
/run/build/vapor/src/backend/openxr.cpp:121:75: error: taking address of temporary array
  121 |     this->glxPbuffer = glXCreatePbuffer(this->xDisplay, fbConfig, (int[]) {
      |                                                                           ^
  122 |         GLX_WIDTH, 16,
      |         ~~~~~~~~~~~~~~                                                     
  123 |         GLX_HEIGHT, 16,
      |         ~~~~~~~~~~~~~~~                                                    
  124 |         None
      |         ~~~~                                                               
  125 |     });
      |     ~
      ```
      
      It seems that it would be valid to pass a temporary array to an `int[n]`, but not `int *`

--- a/src/backend/backend.cpp
+++ b/src/backend/backend.cpp
@@ -15,7 +15,7 @@
 #include <GL/glx.h>
 #include <GL/glext.h>
 
-#include <format>
+#include <string>
 #include <cmath>
 #include <cstring>
 
@@ -117,7 +117,7 @@ Backend::Backend()
                 actionType = XR_ACTION_TYPE_VIBRATION_OUTPUT;
                 break;
         }
-        OpenXR::Action* action = new OpenXR::Action(*this->actionSet, std::format("input_{}", i), std::format("Input {}", i), actionType, {});
+        OpenXR::Action* action = new OpenXR::Action(*this->actionSet, "input_" + std::to_string(i), "Input " + std::to_string(i), actionType, {});
         this->actions.push_back(action);
         OpenXR::ActionSet::ActionBinding binding(*action, inputs[i].path);
         actionSetBindings.push_back(binding);

--- a/src/backend/openxr.cpp
+++ b/src/backend/openxr.cpp
@@ -97,7 +97,7 @@ Session::Session(const Instance& instance, XrSystemId system): instance(instance
         ABORT("Failed to open X11 display");
     }
     int fbConfigsCounts;
-    GLXFBConfig* fbConfigs = glXChooseFBConfig(this->xDisplay, DefaultScreen(this->xDisplay), (int[]) {
+    int configsAttribList[] = {
         GLX_X_RENDERABLE, True,
         GLX_DRAWABLE_TYPE, GLX_PBUFFER_BIT,
         GLX_RENDER_TYPE, GLX_RGBA_BIT,
@@ -107,7 +107,8 @@ Session::Session(const Instance& instance, XrSystemId system): instance(instance
         GLX_ALPHA_SIZE, 8,
         GLX_DEPTH_SIZE, 24,
         None
-    }, &fbConfigsCounts);
+    };
+    GLXFBConfig* fbConfigs = glXChooseFBConfig(this->xDisplay, DefaultScreen(this->xDisplay), configsAttribList, &fbConfigsCounts);
     if (!fbConfigs || fbConfigsCounts == 0)
     {
         ABORT("Failed to find framebuffer configuration");
@@ -118,11 +119,12 @@ Session::Session(const Instance& instance, XrSystemId system): instance(instance
     if (!this->glxContext) {
         ABORT("Failed to create OpenGL context");
     }
-    this->glxPbuffer = glXCreatePbuffer(this->xDisplay, fbConfig, (int[]) {
+    int bufferAttribList[] = {
         GLX_WIDTH, 16,
         GLX_HEIGHT, 16,
         None
-    });
+    };
+    this->glxPbuffer = glXCreatePbuffer(this->xDisplay, fbConfig, bufferAttribList);
     if (!this->glxPbuffer)
     {
         ABORT("Failed to create OpenGL pbuffer");

--- a/src/openvr/impl/utils/pose.h
+++ b/src/openvr/impl/utils/pose.h
@@ -6,6 +6,8 @@
 
 #include "matrix.h"
 
+#include <cstring>
+
 namespace openvr
 {
     namespace utils
@@ -69,7 +71,8 @@ namespace openvr
 
         inline void applyOffset(TrackedDevicePose* trackedDevicePose, const float (&offsetMatrix)[3][4])
         {
-            if (offsetMatrix == (float[3][4]) {{1.0f, 0.0f, 0.0f, 0.0f}, {0.0f, 1.0f, 0.0f, 0.0f}, {0.0f, 0.0f, 1.0f, 0.0f}})
+            static const float identityOffset[3][4] = {{1.0f, 0.0f, 0.0f, 0.0f}, {0.0f, 1.0f, 0.0f, 0.0f}, {0.0f, 0.0f, 1.0f, 0.0f}};
+            if (std::memcmp(offsetMatrix, identityOffset, sizeof(identityOffset)) == 0)
             {
                 return;
             }

